plugins/json/tests/json_04_arrays_of_multi_datatypes.c

```diff
@@ -0,0 +1,95 @@
+		  "    var_string_array:                                                 \n"
+		  "      type: array                                                     \n"
+		  "      size: 2                                                         \n"
+		  "      subtype:                                                        \n"
+		  "        type: array                                                   \n"
+		  "        size: 8                                                       \n"
+		  "        subtype: char                                                 \n"
+		  "                                                                      \n"
+		  "    var_array_array:                                                  \n"
+		  "      type: array                                                     \n"
+		  "      size: 2                                                         \n"
+		  "      subtype: {type: array, size: 3, subtype: int32_t}               \n"
+		  "                                                                      \n"
+		  "    var_records_array:                                                \n"
+		  "      type: array                                                     \n"
+		  "      size: 2                                                         \n"
+		  "      subtype:                                                        \n"
+		  "        type: struct                                                  \n"
+		  "        members:                                                      \n"
+		  "          - var_int32t1: int32_t                                      \n"
+		  "          - var_int32t2: int32_t                                      \n"
+		  "          - var_char: char                                            \n"
+		  "          - var_double: double                                        \n"
+		  "  plugins:                                                            \n"
+		  "    json:                                                             \n"
+		  "      - file: json_04_arrays_of_multi_datatypes.json                  \n"
+		  "        write: [var_string_array, var_array_array, var_records_array] \n";
+
+	PC_tree_t conf = PC_parse_string(CONFIG_YAML);
+	PDI_init(PC_get(conf, ".pdi"));
+
+	/* STRINGS */
+	// char _var_string[] = "string12";
+	// std::string _string = "string12";
+	// PDI_expose("var_string", &_var_string, PDI_OUT);
+
+	char _var_string_array[3][8] = {"string12", "string13", "string14"};
+	PDI_expose("var_string_array", &_var_string_array, PDI_OUT);
+
+	/* ARRAYS */
+	int32_t _var_array_array[2][3] = {{1, 2, 3}, {4, 5, 6}};
+	PDI_expose("var_array_array", _var_array_array, PDI_OUT);
+
+	/* RECORDS */
+	struct record_t {
+		int32_t _var_int32t1;
+		int32_t _var_int32t2;
+		char _var_char; // padding
+		double _var_double;
+	};
+	struct record_t _var_record1 = {14, 15, 'a', 16.};
+	struct record_t _var_record2 = {17, 18, 'b', 19.};
+	// PDI_expose("var_record", &_var_record1, PDI_OUT);
```

Comment:
Do we really want to keep commented-out code?

Reply:
removed in new commit